[UIAsyncTextInput] [WebKit] Adopt API replacements for UITextSuggestionDelegate

whsieh opened the pull request:
#### 1a9127daa810cf548e777868354514f365fd15a7
<pre>
[UIAsyncTextInput] [WebKit] Adopt API replacements for UITextSuggestionDelegate
<a href="https://bugs.webkit.org/show_bug.cgi?id=265097">https://bugs.webkit.org/show_bug.cgi?id=265097</a>
<a href="https://rdar.apple.com/118589483">rdar://118589483</a>

Reviewed by Megan Gardner.

Replace calls to `-[UITextInputSuggestionDelegate setSuggestions:]` with
`-[UIAsyncTextInputDelegate provideCandidateUISuggestions:]`, which provides equivalent
functionality. No change in behavior.

* Source/WebKit/Platform/spi/ios/UIKitSPI.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKFormInputSession invalidate]):
(-[WKContentView updateTextSuggestionsForInputDelegate]):
(-[WKContentView _provideSuggestionsToInputDelegate:]):

Canonical link: <a href="https://commits.webkit.org/271120@main">https://commits.webkit.org/271120@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/afe40d958f498dbf2344ce8a0dbe0a743d3dcd51

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/26875 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/5491 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/28111 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/29086 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/24565 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/7339 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/2882 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/24451 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/27137 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/4314 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/23071 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/3781 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/3836 "Passed tests") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/29721 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/24519 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/24471 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/30066 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/3856 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/2055 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/27976 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/5314 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/6598 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/4323 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/4224 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
```diff
@@ -791,9 +791,7 @@ - (BOOL)requiresStrongPasswordAssistance
-    id <UITextInputSuggestionDelegate> suggestionDelegate = (id <UITextInputSuggestionDelegate>)[_contentView inputDelegate];
-    [suggestionDelegate setSuggestions:nil];
-    _contentView = nil;
+    [std::exchange(_contentView, nil) _provideSuggestionsToInputDelegate:nil];
```
![image](https://github.com/WebKit/WebKit/assets/34530011/cfb696f1-1dcc-4f9e-ad4e-ab971cff78ba)

whsieh committed:
[UIAsyncTextInput] [WebKit] Adopt API replacements for UITextSuggestionDelegate
https://bugs.webkit.org/show_bug.cgi?id=265097
rdar://118589483

Reviewed by Megan Gardner.

Replace calls to `-[UITextInputSuggestionDelegate setSuggestions:]` with
`-[UIAsyncTextInputDelegate provideCandidateUISuggestions:]`, which provides equivalent
functionality. No change in behavior.

* Source/WebKit/Platform/spi/ios/UIKitSPI.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.h:
* Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm:
(-[WKFormInputSession invalidate]):
(-[WKContentView updateTextSuggestionsForInputDelegate]):
(-[WKContentView _provideSuggestionsToInputDelegate:]):

Canonical link: https://commits.webkit.org/271120@main
